ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml

```diff
@@ -16,35 +16,24 @@
 - name: Reload Daemon
   shell: systemctl daemon-reload
 
-- name: Generate modules.dep and map files
-  shell: depmod -a
-  register: depmod_reg
-  ignore_errors: yes
-
-- debug:
-    msg: "{{ depmod_reg }}"
-  when: depmod_reg.rc != 0
-
-- name: Create nbd 128 devices
-  shell: sudo modprobe nbd nbds_max=128
-  register: crt_nbd
-  ignore_errors: yes
-
-- debug:
-    msg: "{{ crt_nbd }}"
-  when: crt_nbd.rc != 0
+- name: Triliovault - set nbd module options in modprobe conf
+  become: true
+  template:
+    src: modprobe-nbd.conf.j2
+    dest: "/etc/modprobe.d/nbd.conf"
 
-- name: Load nbd 128 devices
-  lineinfile:
-    path: /etc/rc.modules
-    line: 'depmod -a && sudo modprobe nbd nbds_max=128'
-    create: yes
-  register: chk_nbd
-  ignore_errors: yes
+- name: Triliovault - set module-load conf for nbd module
+  become: true
+  template:
+    src: module-load-nbd.conf.j2
+    dest: "/etc/modules-load.d/nbd.conf"
 
-- debug:
-    msg: "{{ chk_nbd }}"
-  when: crt_nbd.rc != 0
+- name: Triliovault - load nbd module
+  become: true
+  modprobe:
```

Comment:
Is this tested? Please check if openstack ansible server 'modprobe' ansible module?

Reply:
Yes, it's tested on CentOS8 and deployment worked fine.
I have used the reference: 
https://opendev.org/openstack/openstack-ansible-openstack_hosts/src/commit/6a73496397866c60d1bbbd6f2c30412d18555f91/tasks/configure_metal_hosts.yml

https://docs.ansible.com/ansible/latest/collections/community/general/modprobe_module.html